e2e/suites/provider/cases/vault/vault_mtls.go

```diff
@@ -0,0 +1,99 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+limitations under the License.
+*/
+package vault
+
+import (
+	"github.com/external-secrets/external-secrets-e2e/framework"
+	"github.com/external-secrets/external-secrets-e2e/suites/provider/cases/common"
+	// nolint
+	. "github.com/onsi/ginkgo/v2"
+)
+
+var _ = Describe("[vault] with mTLS", Label("vault-mtls"), func() {
+	f := framework.New("eso-vault-mtls")
+	prov := newVaultProvider(f, true)
+
+	DescribeTable("sync secrets",
+		framework.TableFunc(f, prov),
+		// uses token auth
+		framework.Compose(withTokenAuth, f, common.FindByName, useTokenAuth),
+		framework.Compose(withTokenAuth, f, common.FindByNameAndRewrite, useTokenAuth),
+		framework.Compose(withTokenAuth, f, common.JSONDataFromSync, useTokenAuth),
+		framework.Compose(withTokenAuth, f, common.JSONDataFromRewrite, useTokenAuth),
```

Comment:
These are a lot of tests that need to run, although they don't provide a lot of value. In essence only one test is needed to verify that the mTLS is being accepted by vault.

As mentioned above: if we add one additional listener for vault - one that requires mTLS - then we can add another store which has mTLS enabled and we just need to add one test to the `suites/provider/cases/vault/vault.go` file, something like this:

```go
// ..
framework.Compose(
  withTokenAuthAndMTLS, f, common.FindByName, useMTLSAndTokenAuth)
// ...
```